diff --git a/tests/openonu-go-adapter/Voltha_ONUReconcileTests.robot b/tests/openonu-go-adapter/Voltha_ONUReconcileTests.robot
index 3b34d8f..27a8ce9 100755
--- a/tests/openonu-go-adapter/Voltha_ONUReconcileTests.robot
+++ b/tests/openonu-go-adapter/Voltha_ONUReconcileTests.robot
@@ -338,9 +338,9 @@ Do Reconcile In Omci-Flows-Pushed
     ...    ELSE       Perform Sanity Test    ${suppressaddsubscriber}
     Disable Onu Device
     ${alternativeonustates}=  Create List     omci-flows-deleted
-    Run Keyword If    "${workflow}"=="DT"    Current State Test All Onus    omci-admin-lock
-    ...    ELSE IF    "${workflow}"=="TT"    Current State Test All Onus    omci-admin-lock
-    ...    ELSE       Current State Test All Onus    omci-admin-lock    alternativeonustate=${alternativeonustates}
+    Run Keyword If    "${workflow}"=="DT"    Current State Test All Onus    omci-admin-lock    reqoperstatus=ACTIVE
+    ...    ELSE IF    "${workflow}"=="TT"    Current State Test All Onus    omci-admin-lock    reqoperstatus=ACTIVE
+    ...    ELSE       Current State Test All Onus    omci-admin-lock    reqoperstatus=ACTIVE     alternativeonustate=${alternativeonustates}
     Wait for all ONU Ports in ONOS Disabled    ${ONOS_SSH_IP}    ${ONOS_SSH_PORT}
     Enable Onu Device
     Run Keyword If    "${workflow}"=="DT"    Perform Sanity Test DT     ${suppressaddsubscriber}
